### ISSUE TITLE
Add "validation" guide to verify self-managed installation works

### ISSUE THREAD

**Issue by nikku** [camunda/camunda-platform-docs#2623]
### Describe the problem

As a user I want to install C8 self-managed. To do it, I'll follow one of the existing guides, i.e. the [HELM guide](https://docs.camunda.io/docs/8.0/self-managed/platform-deployment/kubernetes-helm/). 

After the installation I want to ensure that the platform is setup correctly. There is many ways how to validate it (deploy using a modeler, login from different applications, `zbctl`, execute [a shell script](https://camunda.slack.com/archives/C05764N4VNZ/p1695225796410089)), but our documentation is missing clear guidance.

### Describe the desired solution

* Our documentation clearly outlines steps to take to validate a C8 installation works, linking to tools used in the process

### Describe the alternatives considered

* Rather than making this validation a documentation (and manual testing concern) we could bake it into the tools itself, either during initial setup, or using a _system health check_.

----

Related to https://github.com/camunda/camunda-modeler/issues/3808

**Comment by akeller** [comment]
We have a [demo app for Kubernetes](https://docs.camunda.io/docs/next/self-managed/setup/guides/installing-payment-app-example/), but it's part of the Community Hub which feels like an unofficial experience. 

We could also recommend using the [Topology endpoint](https://docs.camunda.io/docs/next/apis-tools/camunda-api-rest/specifications/get-topology/) in the C8 REST API, as I've updated this to replace `zbctl`. 

@hisImminence @theburi, do you have a process that you want to recommend? That way, we can add it consistently across our docs. Ideally, something that is available for all distributions.

**Comment by nikku** [comment]
Thanks @akeller. I think this is relevant ever since. Many solid tools (examples: [Discourse](https://www.discourse.org/), pihole, [matomo](https://matomo.org/)) have this kind of "live" introspection/health endpoints. Thinks can go wrong not only during setup, but also later (re-configuration/upgrading), and being able to refer to it is a huge enabler for successfully setting up, but also confidence in running of a complex software system.

**Comment by hisImminence** [comment]
Good point. I know we have discussions about a "doc" tool that you can run and verify if your stack is healthy. But this is not yet on the roadmap. As Nico said

> There is many ways how to validate it (deploy using a modeler, login from different applications, zbctl, execute [a shell script](https://camunda.slack.com/archives/C05764N4VNZ/p1695225796410089)), but our documentation is missing clear guidance.

Maybe PM has input on what customers might want regarding a "basic validation" or got any customer feedback in this direction?
